packages/meditrak-server/src/routes/importSurveys/ConfigImporter/processConditionConfig.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { BooleanExpressionParser } from '@tupaia/expression-parser';
+import { splitStringOn, splitStringOnComma, translateExpression } from '../../utilities';
+
+export const processConditionConfig = async (models, config) => {
+  const { conditions, defaultValues } = config;
+  return {
+    conditions: await translateConditions(models, conditions, defaultValues),
+  };
+};
+
+/**
+ * Translate and combine conditions and defaultValues into 1 config object.
+ *
+ * Excel config:
+ * conditions: Yes:RHS6UNFPA1353Calc >= 3,No:RHS6UNFPA1353Calc < 3
+ * defaultValues: Yes.RHS6UNFPA1353Calc:0,No.RHS6UNFPA1353Calc:0
+ *
+ * will be converted to:
+ *
+ * conditions: {
+ *   Yes: {
+ *     formula: '5fa0cdcd5da1e614f30001fd >= 3',
+ *     defaultValues: {
+ *        '5fa0cdcd5da1e614f30001fd': '0',
+ *     },
+ *   },
+ *   No: {
+ *     formula: '5fa0cdcd5da1e614f30001fd < 3',
+ *     defaultValues: {
+ *       '5fa0cdcd5da1e614f30001fd': '0',
+ *     },
+ *   },
+ * }
+ * @param {*} models
+ * @param {*} conditionsConfig
+ * @param {*} defaultValuesConfig
+ */
+const translateConditions = async (models, conditionsConfig, defaultValuesConfig = '') => {
+  const conditions = splitStringOnComma(conditionsConfig);
+  const translatedConditions = {};
+  const expressionParser = new BooleanExpressionParser();
+
+  // Translate the expressions
+  for (const condition of conditions) {
+    const [targetValue, expression] = splitStringOn(condition, ':');
+    const codes = expressionParser.getVariables(expression);
```

Comment:
What if the data element code starts with a number? Do any currently?

Reply:
Answered that myself. A few do, the ones representing universal drug codes

<img width="566" alt="Screen Shot 2020-11-13 at 3 34 35 PM" src="https://user-images.githubusercontent.com/1274422/99021574-d35ffb80-25c5-11eb-929e-c7084e6f0a23.png">

Reply:
Yeah, so in the excel config, users will have to prefix $ to the variables then...